src/EnergyPlus/EvaporativeCoolers.cc

```diff
@@ -4041,6 +4048,20 @@ void CalcZoneEvaporativeCoolerUnit(EnergyPlusData &state,
     {
         auto &zoneEvapUnit = state.dataEvapCoolers->ZoneEvapUnit(UnitNum);
 
+        Real64 relativeHumidity = 100.0 * Psychrometrics::PsyRhFnTdbWPb(state,
+                                                                        state.dataLoopNodes->Node(zoneEvapUnit.ZoneNodeNum).Temp,
+                                                                        state.dataLoopNodes->Node(zoneEvapUnit.ZoneNodeNum).HumRat,
+                                                                        state.dataEnvrn->OutBaroPress,
+                                                                        "CalcZoneEvaporativeCoolerUnit");
+        if (relativeHumidity > zoneEvapUnit.ShutOffRelativeHumidity) {
+            // unit is off when humidity is too high
+            PartLoadRatio = 0.0;
+            zoneEvapUnit.UnitPartLoadRatio = PartLoadRatio;
+            CalcZoneEvapUnitOutput(state, UnitNum, PartLoadRatio, SensibleOutputProvided, LatentOutputProvided);
+            zoneEvapUnit.IsOnThisTimestep = false;
+            return;
```

Comment:
I assume this is fine.  As long as all the unit outputs and node connections are cleaned up, then we are able to return early.  I only bring it up because returning early is a common source of bugs in component models.  In this case, since you are calling Calc, I think it should be OK.

Reply:
is it better to keep a flag and return at the end? I'll take a note here and do that in the future. @Myoldmopar

Reply:
It just depends.  In the past, returning early was a common way for nodes to not get updated, leading to many bugs in E+.  Those days are mostly behind us, but returning early is still something that jogs the memory.  As long as the component "finishes" its work, then go ahead and return, no problem.